src/vcf/record.cpp

```diff
@@ -548,7 +559,7 @@ namespace ebi
             }
         } else if (type == "Flag") {
             int numeric_value = std::stoi(value);
-            if (numeric_value != 0 && numeric_value != 1) {
+            if (value.size() > 1 || (numeric_value != 0 && numeric_value != 1)) {
                 message = " (a flag value must be \"0, 1 or none\")";
```

Comment:
Added this because we were accepting values like 0.34 or 1.89 (as they get rounded off to 0 & 1) for Flag which are erroneous.